internal/sync/controller.go

```diff
@@ -177,8 +177,13 @@ func (p Controller) diffEvents(sourceEvents []models.Event, sinkEvents []models.
 
 		switch {
 		case !exists:
+			if event.Metadata.SourceID != "p.sink.GetSinkID()" {
```

Comment:
The quotes are wrong. Currently that condition is always true.

Shouldn't the check be `event.Metadata.SourceID == p.sink.GetSinkID()`? Syncing eventA (from calendar A) yields eventACopy (calendar B, SourceID = calendar A). After deleting eventA, `case !exists` applies when processing eventACopy, and `p.sink` would be calendar A.

Reply:
yes true but at least currently there is no p.sink.GetSinkID() that's why i marked it as string.
For the actual bug please take a look at the linked issue :)